pkg/controller/mutators/core/controller_test.go

```diff
@@ -194,5 +228,103 @@ func TestReconcile(t *testing.T) {
 		}, timeout).Should(gomega.Succeed())
 	})
 
+	t.Run("Conflicting mutators are marked not enforced and conflicts can be resolved", func(t *testing.T) {
+		mFoo := newAssign("foo", "spec.foo", "value-1")
+		mFooID := types.MakeID(mFoo)
+		mBar1 := newAssign("bar-1", "spec.bar[name: bar].qux", "value-2")
+		mBar1ID := types.MakeID(mBar1)
+		mBar2 := newAssign("bar-2", "spec.bar.qux", "value-3")
+		mBar2ID := types.MakeID(mBar2)
+
+		g.Expect(c.Create(ctx, mFoo.DeepCopy())).NotTo(gomega.HaveOccurred())
+		g.Expect(c.Create(ctx, mBar1.DeepCopy())).NotTo(gomega.HaveOccurred())
+		g.Expect(c.Create(ctx, mBar2.DeepCopy())).NotTo(gomega.HaveOccurred())
+
+		g.Eventually(func() error {
+			err := podStatusMatches(ctx, c, pod, mFooID, hasStatusErrors(nil))
+			if err != nil {
+				return err
+			}
+
```

Comment:
Will this fail sometimes?

we create three mutators (two of which are in conflict) and expect that our initial assertion (`hasStatusErrors(nil)`) will always happens successfully.  Would it make more logical sense to:

1. Apply the two non-conflicting mutators
2. Assert no status errors
3. Apply the conflicting mutator
4. Assert that status error exists for both conflicting mutators
5. remove one of the conflicting mutators
6. assert there are no more errors

Reply:
Done.